Draft-Accepted/RFCXXXX-Command-Shell-Integration.md

```diff
@@ -0,0 +1,366 @@
+The user would simply install the tool and it would automatically be integrated with the shell if the session is interactive.
+
+PowerShell 7 would emit on startup any automatically loaded extensions unless `-NoLogo` is specified.
+
+### Structured Output with Formatting
+
+```powershell
+az group list
+```
+
+Current output:
+
+```output
+[
+  {
+    "id": "/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/RG1",
+    "location": "westcentralus",
+    "managedBy": null,
+    "name": "RG1",
+    "properties": {
+      "provisioningState": "Succeeded"
+    },
+    "tags": {
+      "RequestID": "1111"
+    },
+    "type": "Microsoft.Resources/resourceGroups"
+  },
+  {
+    "id": "/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/RG2",
+    "location": "westcentralus",
+    "managedBy": null,
+    "name": "RG2",
+    "properties": {
+      "provisioningState": "Succeeded"
+    },
+    "tags": {
+      "RequestID": "1111"
+    },
+    "type": "Microsoft.Resources/resourceGroups"
+  }
+]
+```
+
+In PowerShell, this output can be piped to `ConvertFrom-Json` and it's an additional step for every `az` command execution.
+It would be preferrable to have the tool output structured data in a format that PowerShell can automatically convert to an object
+and used similarly to a cmdlet but only if the `$typeNames` property is set (see below).
+
+Proposed output with custom formatting showing the most relevant information as well as a nested property:
+
+```output
+id                                                                                       location      name                  provisioningState
+--                                                                                       --------      ----                  ----------
+/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/ERNetwork-PvtApp      westcentralus RG1                   Succeeded
+/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/ERNetwork-SVC         westcentralus RG2                   Succeeded
+```
+
+### Tab Completion
+
+```powershell
+az g<tab>
+```
+
+This should result in:
+
+```powershell
+az group
+```
+
+As the tab completer would automtically be discovered by PowerShell and used.  In this example, there is only one subcommand starting with `g`.
+
+### Help
+
+```powershell
+get-help az
+```
+
+The help content would preferrably be in markdown with advanced rendering available in the shell.
+Normal help searches would also look within the help content defined by the tool.
+
+### Predictors and Feedback Providers
+
+PowerShell has an extensibility model for Predictors and Feedback Providers.
+Typically, these are implemented as modules with a cmdlet to register them with the PowerShell engine.
+In this case, the extension would still need to be managed code, but would be automatically discovered and loaded by the shell.
+Since the user installed the tool, the expectation is they want to use it.
```

Comment:
Is this statement is exclusive to tools that hope to register with feedbackprovider and or predictors?